src/main/java/com/hubspot/imap/ImapClientFactory.java

```diff
@@ -56,34 +60,27 @@ public ImapClientFactory(ImapClientFactoryConfiguration configuration, KeyStore
   }
 
   public CompletableFuture<ImapClient> connect(String clientName, ImapClientConfiguration clientConfiguration) {
-    final SslContext finalSslContext;
-    if (clientConfiguration.trustManagerFactory().isPresent()) {
-      try {
-        finalSslContext = SslContextBuilder.forClient()
-            .trustManager(clientConfiguration.trustManagerFactory().get())
-            .build();
-      } catch (SSLException e) {
-        throw new RuntimeException(e);
-      }
-    } else {
-      finalSslContext = sslContext;
-    }
+    Supplier<SslContext> sslContextSupplier = Suppliers.memoize(() -> getSslContext(clientConfiguration));
+    boolean useSsl = clientConfiguration.useSsl() && !clientConfiguration.proxyConfig().isPresent();
```

Comment:
There's a little sleight of hand here:
- The proxy protocol we're using doesn't support SSL. Nor would we want SSL since we'd negotiate against the proxy server and not the downstream server
- If SSL is turned on, we'll turn on SSL *after* we tell the Proxy to redirect us to the remote host here https://github.com/HubSpot/NioImapClient/pull/62/files#diff-4a457cc0f43414006b3ba95d31ca7cebR119